hydra-node/test/Hydra/Chain/Direct/StateSpec.hs

```diff
@@ -230,6 +237,13 @@ spec = parallel $ do
         UnsupportedLegacyOutput{} -> property True
         _ -> property False
 
+    prop "reject Commits with more than 100 ADA" $
+      forAllCommitWithMoreThan100ADA $ \case
+        FailedToPostTx{failureReason} ->
+          property $
+            failureReason == "hard-coded limit of 100 ADA for the commit tx on mainnet reached."
```

Comment:
Should: Not match error strings.

This is another reason why a dedicated data constructor similar like `UnsupportedLegacyOutput` should be used.